dev/documentation/en/utilities-hooks.md

```diff
@@ -0,0 +1,159 @@
+---
+title: Hooks - Technical Documentation - OJS|OMP|OPS
+---
+
+# Hooks
+
+> Hooks are similar to filters and actions in WordPress.
+{:.tip}
+
+Hooks allow plugins, themes, and third-party code to intervene in the application and make modifications.
+
+Use `HookRegistry` to create a hook.
+
+```php
+HookRegistry::call('Context::delete', [$context]);
+```
+
+Register a callback function to the hook from anywhere in the code, including a third-party plugin. The example below sends an email whenever a journal, press or pre-print server is deleted.
+
+```php
+HookRegistry::register('Context::delete', function($hookName, $args) {
+	$context = $args[0];
+
+	// Code below is for demonstration and does not work
+	Email::send('admin@admin.com', 'The journal ' . $context->getLocalizedName() . ' was deleted.');
+});
+```
+
+It's also possible to use a hook to modify data. In the example below, every new journal that is created is assigned a default `publisherInstitution`.
+
+```php
+HookRegistry::call('Context::add', [$context, $request]);
+```
+
+```php
+HookRegistry::register('Context::add', function($hookName, $args) {
+	$context = $args[0];
+
+	$context->setData('publisherInstitution', 'Public Knowledge Project');
+});
+```
+
+By default, PHP only uses references when passing objects between functions. Pass variables into hooks by reference (`&`) when they are not an object.
+
+```php
+$xml = '<xml ...>';
+HookRegistry::call('Deposit::xml', [&$xml]);
+```
+
+```php
+HookRegistry::register('Deposit::xml', function($hookName, $args) {
+	$xml =& $args[0];
+
+	// $xml is now a reference to the same string instead of a copy
+	// so changes made here will effect the original variable
+});
```

Comment:
Hook functions should *always* return a value, generally `false`. I'd suggest consistently adding a return value here to guide good practices.

Reply:
In fact, a short section on hook returns might be useful. A good motivating example might be `Mail::send`: if a hook callback returns `false`, further registrants continue to be called in order; however, if a `Mail::send` callback returns `true`, then later registrants *are not* called, and the email is not sent.

If you want e.g. to augment an email by adding a CC recipient to each message using a plugin, then you'll want to modify the Mail object to add the recipient, then return `false`.

However, if you want to use a plugin to integrate OJS with another mail delivery tool like Mailchimp, you'll want to deliver the message via Mailchimp in your plugin callback, then return `true` to prevent OJS from trying to deliver the message via the normal means. (And you'll want to register the callback late, so that any other plugin that wants to alter the email can do so first.)